src/uct/api/uct.h

```diff
@@ -1202,6 +1202,8 @@ struct uct_md_attr {
         uint64_t             flags;     /**< UCT_MD_FLAG_xx */
         uint64_t             reg_mem_types; /**< Bitmap of memory types that Memory Domain can be registered with */
         uint64_t             detect_mem_types; /**< Bitmap of memory types that Memory Domain can detect if address belongs to it */
+        uint64_t             detect_mem_attr; /**< Bitmap of memory types on which MD can detect memory
```

Comment:
how is it different than detect_mem_types?

Reply:
`detect_mem_types` can only tell what kind of memory type (CUDA, ROCM, HOST, etc) the buffer pointer belongs to. `detect_mem_attr` returns memory type and device backing the pointer. Eventually, we can deprecate `detect_mem_types` if needed.

Reply:
at this time, we should extend existing function in case of UCT API, not add new one

Reply:
I maybe missing something. I read this previous comment from you as extending the API - https://github.com/openucx/ucx/pull/4974#discussion_r408745518
Did I misinterpret? @yosefe Can you please clarify?

Reply:
there should be one flag per function
so in prev comment i've advised to add new flag, but thinking of it more, IMHO it's better replace the existing function and the existing flag

Reply:
If I remember correctly, the first PR I had proposed was along the lines of replacement but the recommendation then was to add a new one and deprecate the old use piece-wise to limit the changes per PR. Replacing still means that there would have to be changes across UCP/UCT. I'm not opposed to either this approach or replacing flag/function.

Reply:
Hi @yosefe Any further thoughts on this or should we add changes to replace existing function/flags?